lib/schema-compilers.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+const Ajv = require('ajv')
+const fastJsonStringify = require('fast-json-stringify')
+
+function ValidatorSelector () {
+  const validatorPool = new Map()
+
+  return (externalSchemas, options, cache) => {
+    const externals = JSON.stringify(externalSchemas)
+    const ajvConfig = JSON.stringify(options.customOptions)
```

Comment:
Why don’t you clear the cache for the current plugin if the relevant set methods are called?

Reply:
Do you mean that the cache could be moved from input to a local variable?
(if so I agree: I will move it to isolate better this behaviour)

Reply:
I mean that you can treat it as every other encapsulated properties.

Reply:
Done